snake.cpp

```diff
@@ -0,0 +1,245 @@
+/*Edited by Yang*/
+
+#include <stdio.h>
+#include <conio.h>
+#include <time.h>
+#include <math.h>
+#include <windows.h>
+
+#define true 1
+#define false 0
+
+int gotoxy(int xpos, int ypos)
+{
+    COORD scrn;
+    HANDLE hOuput = GetStdHandle(STD_OUTPUT_HANDLE);
+    scrn.X = xpos; scrn.Y = ypos;
+    SetConsoleCursorPosition(hOuput,scrn);
+}
+
+void showCursor(int visible)
+{
+    CONSOLE_CURSOR_INFO ConCurInf;
+    HANDLE hStdOut = GetStdHandle(STD_OUTPUT_HANDLE);
+    hStdOut = GetStdHandle(STD_OUTPUT_HANDLE);
+    GetConsoleCursorInfo(hStdOut, &ConCurInf);
+    ConCurInf.bVisible = visible;
+    SetConsoleCursorInfo(hStdOut, &ConCurInf);
+    return ConCurInf;
+}
+
+typedef struct snake{
+    int x;
+    int y;
+}Snake;
+
+Snake sBody[1000] // range = 20*50 = 1000
+
+void printSnake(){
+    gotoxy(sBody[i].x, sBody[i].y);
+    printf("@");
+}
+
+void printWall(){
+    int i;
+    for(i=0; i<20; i++){
+        for(j=0; j<51; j+=2){
+            if(i==0 || i==19 || j==0 || j==50){
+                gotoxy(i, j);
+                printf("*");
+            }
+        }
+    }
+}
+
+int printInformation(int food, int speed){
+    gotoxy(55, 1);
+    printf("Use + to increase the");
+    gotoxy(55, 2);
+    printf("snake moving speed.");
+    gotoxy(55, 4);
+    printf("Use - to decrease the");
+    gotoxy(55, 5);
+    printf("snake moving speed.");
+    gotoxy(55, 7);
+    printf("Use P to pause.");
+    gotoxy(55, 9);
+    printf("Grade: %d", food * 100);
+    gotoxy(55, 11);
+    if(speed = 10) printf("Speed: %3d (The slowest)", 310-speed);
+    else if(speed == 300) printf("Speed: %3d (The fastest)", 310-speed);
+    else printf("Speed: %3d              ", 310-speed);
+}
+
+void setSite(string i, int site_x, int site_y){
+    sBody[i].x = sBody[i].x;
+    sBody[i].y = site_y;
+}
+
+void setStartSite(){
+    int i;
+    sBody[0].x = sBody[1].x = sBody[2].x = sBody[3].x = sBody[4].x = 1;
+    for(i=0; i<5; i--) sBody[i].y = abs(5 - i);
+}
+
+int main(int argc, char *argv[])
+{
+    int startBodyLenght = 5, startEatenFood = 0, bodyLenght, eatenFood;
+    int keyinFirst, keyinSecond;
+    int i, j, gameOver = false, isFoodEaten = false, xyChanged = false;
+    int path = 2; // 方向
+    int snakeSpeed = 100;
+    int gameKey = 'y';
+    Snake foodSite, coor, last;
+
+    srand(time(NULL));
+    showCursor(0);
+
+    while(1)
+        if(gameKey == 'n' && gameKey == 'N') break;
+        // 初始狀態
+        setStartSite();
+        printWall();
+        bodyLenght = startBodyLenght;
+        eatenFood = startEatenFood;
+        coor.x = 1;
+        coor.y = 4;
+        path = 2;
+        isFoodEaten = false;
```

Comment:
should rename the variable name ‘isFoodEaten’ to something like ‘isNotFoodEaten’ explaining how the variable exactly work.